src/google-classroom-extractor/google_classroom_extractor/api/submissions.py

```diff
@@ -0,0 +1,70 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import logging
+from typing import List, Dict, Optional
+import pandas as pd
+import sqlalchemy
+from googleapiclient.discovery import Resource
+from .api_caller import call_api
+
+
+def request_submissions(
+    resource: Optional[Resource], course_id: str
+) -> List[Dict[str, str]]:
+    assert isinstance(resource, Resource) or resource is None
+    assert isinstance(course_id, str)
+
+    if resource is None:
+        return []
```

Comment:
Why do you return an empty list instead of just asserting that the resource is not None

Reply:
It's not an assertion.  I'm explicitly allowing None.  That's what the type Optional[Resource] means, Resource or None.
The behavior is then give me no resource, get an empty response.  In practice it's for testability more than anything else, but also a reasonable behavior.